### ISSUE TITLE
`VimResized` should be added by default in the `trigget_events` for the `register`

### ISSUE THREAD

**Issue by pocco81** [kwkarlwang/bufresize.nvim#5]
Hello :) First off, I really love this plugin! not only because of how useful it is, but also because it's all I ever needed to close [this issue](https://github.com/Pocco81/TrueZen.nvim/issues/53) in a repo of mine.

Now, why should `VimResized` be added by default? Here is an example:

### Without `VimResized`

https://user-images.githubusercontent.com/58336662/134844262-fdab2a94-008f-4f7c-8cdd-a6e6b2daf464.mp4

### With `VimResized`

https://user-images.githubusercontent.com/58336662/134844265-03f6e928-c552-4247-94f8-96444cbae39f.mp4

## Explanation

Basically what's going on here is that I have `tmux` open, with 3 splits, and the one in the left is being focused at the beginning and then unfocused. Without that event, after one moves the cursor (in my case), the text "shifts" (I have `wrap` set), however with `VimResized` it does not happens, for obvious reasons.

Notice that this isn't like a `tmux`-only thing, you can test this by simply using any other terminal multiplexer or by resizing the terminal.

Perhaps it'd be good it add it by default?

**Comment by kwkarlwang** [comment]
Hi Pocco81, thank you for the kind words! Glad that the plugin has helped you out.

Now I am not too sure I have reproduced correctly. But I have three tmux windows opened, with `wrap` turned on, and I am toggling the tmux zoom on my left buffer and everything seems to work correctly, I was not able to reproduce the error.

My setup is as followed
```lua
			require("bufresize").setup({
				register = {
					keys = {
						{ "n", "<leader>w<", "30<C-w><", NS },
						{ "n", "<leader>w>", "30<C-w>>", NS },
						{ "n", "<leader>w+", "10<C-w>+", NS },
						{ "n", "<leader>w-", "10<C-w>-", NS },
						{ "n", "<leader>w_", "<C-w>_", NS },
						{ "n", "<leader>w=", "<C-w>=", NS },
						{ "n", "<leader>w|", "<C-w>|", NS },
						{ "n", "<leader>wo", "<C-w>|<C-w>_", NS },
						{ "", "<LeftRelease>", "<LeftRelease>", NS },
						{ "i", "<LeftRelease>", "<LeftRelease><C-o>", NS },
					},
				},
			})

```

I suspect there is actually a Vim error, can you check `:messages` after resizing and see if there are any errors?

https://user-images.githubusercontent.com/38927155/134849663-7d651cef-6ebf-4c43-a915-5bf5eb390a1a.mov

**Comment by kwkarlwang** [comment]
I don't know why adding `VimResized` to `register` help you solve the problem, because if you have `VimResized`  in `resize`, then the [resize](https://github.com/kwkarlwang/bufresize.nvim/blob/ecd2aac2d97134e8fc5065af70680905db4639dd/lua/bufresize.lua#L76) actually call `register` to record the state after the applying the proportional resize to the windows.

**Comment by pocco81** [comment]
Huh that's weird. Can I see your bufresize conf? Mine is in the `dev` branch on my dots repo.

**Comment by kwkarlwang** [comment]
For sure, it is [here](https://github.com/kwkarlwang/dotfiles/blob/9edbe821d8f67e73de2a4570e0626f951eff7679/vim/.config/nvim/lua/plugins.lua#L514).

I have tried the config on your dev branch with the same lazy loading event, commit version and setup, but still cannot reproduce the error. 

Now I have encountered needing to press one more key for the wrapping to happen, but then later I found out that actually means there is an error occurred in the command window, it is just not shown when resized, you would have to manually check it through `:messages`. Not sure if this is the same case as yours.
